flink-baseline/flink-jdbc-sink/src/main/java/com/ness/flink/sink/jdbc/connector/JdbcConnectionProvider.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020-2023 Ness USA, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ness.flink.sink.jdbc.connector;
+
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.SQLException;
+import org.apache.flink.annotation.Internal;
+
+/**
+ * @author Khokhlov Pavel
+ */
+@Internal
+public interface JdbcConnectionProvider extends Serializable {
+    /**
+     * Check whether possible existing connection is valid or not through {@link
+     * Connection#isValid(int)}.
+     *
+     * @param connection jdbc connection
+     * @return true if existing connection is valid
+     * @throws SQLException sql exception throw from {@link Connection#isValid(int)}
+     */
+    boolean isConnectionValid(Connection connection) throws SQLException;
```

Comment:
not quite clear what this interface is for. It is called `Provider`, but its operations are for a connection passed from outside, as in Util class

Reply:
Hi!
Thanks for feedback.
Let me explain.
I would like to keep at least names of original flink-connector-jdbc classes.
See
https://github.com/apache/flink-connector-jdbc/blob/58b0eea7286e01c505f2b64f744df7784a9a6273/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/connection/JdbcConnectionProvider.java#L45

Original (community) Interface was designed to keep State of Connection.
I don't want to do it because it brings a lot of complexity (based on my experience on Instinet project) that's why I decided to simplify it.
Implementation validation for Connection could be different that's why we have `SimpleJdbcConnectionProvider`

Hope I answered on your question.

Reply:
ok. I personally don't like the approach where arguments are passed explicitly. I'd like to see a Provider interface with only one method, `getConnection()`, everything else should be internal

Reply:
I agree, but I decided to keep Original name of Interface. 
It will help me track changes with Community edition version

Reply:
done. see my latest commits